### PR TITLE
E2E: remove assertion from ACL role test

### PR DESCRIPTION
**Pull request by tgross**
The ACL role test asserts that the role has various permissions by listing jobs in namespaces. It never creates jobs, because we can make all the assertions we need by checking the error. But the test included an assertion that the namespace was empty. Usually this will be the case, but if the previous test case has not completed its GC (which is sync), then it's possible a stopped job will be in the namespace. Because this assertion is irrelevant for this test, remove it.

Fixes: https://github.com/hashicorp/nomad/issues/19109

### CODE REVIEW

File: e2e/acl/acl_role_test.go [hunk 1]
```diff
@@ -98,12 +98,11 @@ func testACLRole(t *testing.T) {
 	customNSQueryMeta := api.QueryOptions{Namespace: ns.Name, AuthToken: aclTokenCreateResp.SecretID}
 	defaultNSQueryMeta := api.QueryOptions{Namespace: "default", AuthToken: aclTokenCreateResp.SecretID}
 
-	jobListResp, _, err := nomadClient.Jobs().List(&customNSQueryMeta)
-	require.NoError(t, err)
-	require.Empty(t, jobListResp)
```
Comment (gulducat) [inline e2e/acl/acl_role_test.go:103]: this is the offending assertion, yeah?

File: e2e/acl/acl_role_test.go [hunk 2]
```diff
@@ -122,35 +121,34 @@ func testACLRole(t *testing.T) {
...
-	require.Len(t, aclRoleUpdateResp.Policies, 2)
+	must.NoError(t, err)
+	must.Eq(t, aclRoleCreateResp.ID, aclRoleUpdateResp.ID)
+	must.Len(t, 2, aclRoleUpdateResp.Policies)
 
 	// Try listing the jobs in the default namespace again to ensure we now
 	// have permission due to the updated role.
-	jobListResp, _, err = nomadClient.Jobs().List(&defaultNSQueryMeta)
-	require.NoError(t, err)
-	require.Empty(t, jobListResp)
```
Comment (gulducat) [inline e2e/acl/acl_role_test.go:133]: and here?

**Review by gulducat — APPROVED** [review]

File: e2e/acl/acl_role_test.go (hunk 1 above)
Comment (tgross) [reply e2e/acl/acl_role_test.go:103]: Yes!

File: e2e/acl/acl_role_test.go (hunk 2 above)
Comment (tgross) [reply e2e/acl/acl_role_test.go:133]: Yes!

**Pull request merged**